src/ILLink.CodeFix/DAMCodeFixProvider.cs

```diff
@@ -0,0 +1,138 @@
+			} else if (diagnostic.Id == DiagnosticId.DynamicallyAccessedMembersMethodAccessedViaReflection.AsString ()) {
+				return new[] { syntaxGenerator.AttributeArgument (syntaxGenerator.BitwiseOrExpression (syntaxGenerator.DottedName ("System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.NonPublicMethods"), syntaxGenerator.DottedName ("System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes.PublicMethods"))) };
+			} else if (diagnostic.Id == DiagnosticId.DynamicallyAccessedMembersMismatchParameterTargetsThisParameter.AsString () || diagnostic.Id == DiagnosticId.DynamicallyAccessedMembersMismatchFieldTargetsThisParameter.AsString () || diagnostic.Id == DiagnosticId.DynamicallyAccessedMembersMismatchParameterTargetsMethodReturnType.AsString ()) {
+				return new[] { syntaxGenerator.AttributeArgument (syntaxGenerator.TypedConstantExpression (targetSymbol.GetAttributes ().First (attr => attr.AttributeClass?.ToDisplayString () == DynamicallyAccessedMembersAnalyzer.FullyQualifiedDynamicallyAccessedMembersAttribute).ConstructorArguments[0])) };
+			} else {
+				return new[] { syntaxGenerator.AttributeArgument (syntaxGenerator.LiteralExpression ("")) };
+			}
+		}
+
+		public sealed override FixAllProvider GetFixAllProvider ()
+		{
+			// See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+			return WellKnownFixAllProviders.BatchFixer;
+		}
+
+		protected static async Task BaseRegisterCodeFixesAsync (CodeFixContext context)
+		{
+
+			bool ReturnAttribute = false;
+			var document = context.Document;
+			var root = await document.GetSyntaxRootAsync (context.CancellationToken).ConfigureAwait (false);
+			var diagnostic = context.Diagnostics.First ();
+			var props = diagnostic.Properties;
+			var model = await document.GetSemanticModelAsync (context.CancellationToken).ConfigureAwait (false);
+			SyntaxNode diagnosticNode = root!.FindNode (diagnostic.Location.SourceSpan);
+			var attributableSymbol = (diagnosticNode is InvocationExpressionSyntax { 
+			    Expression: MemberAccessExpressionSyntax { Expression: IdentifierNameSyntax name } }) 
+					? model.GetSymbolInfo (name).Symbol 
+					: null;
+
+			if (attributableSymbol is null) return;
+
+			var attributableNodeList = attributableSymbol.DeclaringSyntaxReferences;
+
+			var attributableNode = (attributableNodeList.Length == 1) ? attributableNodeList[0].GetSyntax () : null;
+
+			if (attributableNode is null) return;
+			var targetSymbol = model!.GetSymbolInfo (diagnosticNode).Symbol!;
+			var attributeSymbol = model!.Compilation.GetTypeByMetadataName (FullyQualifiedAttributeName)!;
+			var attributeArguments = GetAttributeArguments (attributableSymbol, targetSymbol, SyntaxGenerator.GetGenerator (document), diagnostic);
+			var codeFixTitle = CodeFixTitle.ToString ();
+			var returnAttributes = AttriubuteOnReturn ();
+			if (returnAttributes.Contains (attributeSymbol.ToString ())) {
```

Comment:
This had faulty logic - I've updated it now so it's if the `returnAttributes` contains the `diagnostic.Id`, because certain diagnostics always have attributes on the return type (is my understanding). This isn't implemented yet, but would be necessary to have when we do implement fixers for these DiagnosticIds. Is it best practice to leave it in or should I take it out and add it back later?

Reply:
I'll take it out, and we can add it back in if we decide to support the return diagnostics later on